--- /home/chris/oh_patchlibc_0803/openharmony/third_party/musl/src/process/waitpid.c	2023-08-07 15:19:23.280403360 +0800
+++ topatch/src/process/waitpid.c	2023-08-07 16:24:49.180447946 +0800
@@ -1,7 +1,21 @@
+/*
+ * Copyright (c) 2023 Institute of Parallel And Distributed Systems (IPADS), Shanghai Jiao Tong University (SJTU)
+ * Licensed under the Mulan PSL v2.
+ * You can use this software according to the terms and conditions of the Mulan PSL v2.
+ * You may obtain a copy of Mulan PSL v2 at:
+ *     http://license.coscl.org.cn/MulanPSL2
+ * THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND, EITHER EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT, MERCHANTABILITY OR FIT FOR A PARTICULAR
+ * PURPOSE.
+ * See the Mulan PSL v2 for more details.
+ */
+
 #include <sys/wait.h>
 #include "syscall.h"
 
+#include <chcore/proc.h>
+
 pid_t waitpid(pid_t pid, int *status, int options)
 {
-	return syscall_cp(SYS_wait4, pid, status, options, 0);
+    return chcore_waitpid(pid, status, options, 0);
 }
