diff --git a/drivers/Kconfig b/drivers/Kconfig
index 2552b32..83c4f4a 100644
--- a/drivers/Kconfig
+++ b/drivers/Kconfig
@@ -248,4 +248,6 @@ source "drivers/rknpu/Kconfig"
 
 source "drivers/accesstokenid/Kconfig"
 
+source "drivers/tzdriver/Kconfig"
+
 endmenu
diff --git a/drivers/Makefile b/drivers/Makefile
index 9bce617..4766321 100644
--- a/drivers/Makefile
+++ b/drivers/Makefile
@@ -197,3 +197,4 @@ obj-$(CONFIG_RK_HEADSET)	+= headset_observe/
 obj-$(CONFIG_RK_NAND)		+= rk_nand/
 obj-$(CONFIG_ROCKCHIP_RKNPU)	+= rknpu/
 obj-$(CONFIG_ACCESS_TOKENID)	+= accesstokenid/
+obj-$(CONFIG_TZDRIVER)         += tzdriver/
\ No newline at end of file
diff --git a/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dtsi b/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dtsi
index 9b22698..db2e4f4 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3588-orangepi-5-plus.dtsi
@@ -241,6 +241,11 @@
 		regulator-boot-on;
 		regulator-always-on;
 	};
+
+	trusted_core {
+		compatible = "trusted_core";
+		interrupt = <0 296 4>;
+	};
 };
 
 &backlight {
